Lib/test/test_ntpath.py

```diff
@@ -422,15 +422,20 @@ def test_ismount(self):
             self.assertTrue(ntpath.ismount(b"\\\\localhost\\c$"))
             self.assertTrue(ntpath.ismount(b"\\\\localhost\\c$\\"))
 
+    def assertEqualCI(self, s1, s2):
+        """Assert that two strings are equal ignoring case differences
+        """
+        self.assertEqual(s1.lower(), s2.lower())
+
     @unittest.skipUnless(nt, "OS helpers require 'nt' module")
     def test_nt_helpers(self):
         # Trivial validation that the helpers do not break, and support both
         # unicode and bytes (UTF-8) paths
 
         drive, path = ntpath.splitdrive(sys.executable)
         drive = drive.rstrip(ntpath.sep) + ntpath.sep
-        self.assertEqual(drive, nt._getvolumepathname(sys.executable))
-        self.assertEqual(drive.encode(),
+        self.assertEqualCI(drive, nt._getvolumepathname(sys.executable))
```

Comment:
This test assumes that `sys.executable` is on the same volume as the path's drive. Say we have a volume mounted at "C:\Mount\Work" and the Python executable is "C:\Mount\Work\CPython\PCbuild\amd64\python_d.exe". In this case `drive` (actually the drive root) is `"C:\\"`, and `nt._getvolumepathname(sys.executable)` is `"C:\\Mount\\Work\\"`. 

The unresolved executable path also might contain multiple junctions or symlinks, such that `_getvolumepathname` returns a path with a different drive. Probably we should be resolving links in `sys.executable` at startup, which IIRC POSIX already does. 

Also, the test assumes the file-system encoding is UTF-8 instead of encoding the path with `os.fsencode`. Python may be configured to use the legacy ANSI encoding. Only recent builds of Windows 10 support setting the ANSI codepage as UTF-8. 

The following should address these concerns:

    executable = nt._getfinalpathname(sys.executable)

    for path in executable, os.fsencode(executable):
        volume_path = nt._getvolumepathname(path)
        path_drive = ntpath.splitdrive(path)[0]
        volume_path_drive = ntpath.splitdrive(volume_path)[0]
        self.assertEqualCI(path_drive, volume_path_drive)

Reply:
Thanks; I've adopted your code